src/app.rs

```diff
@@ -261,6 +263,9 @@ impl App {
 
         let (processed, batching_builder) = processed_builder.seal_and_continue();
         let (batching, mut latest_builder) = batching_builder.seal_and_continue();
+
+        // We are duplicating updates here for some commitments that were batched but
```

Comment:
Are though? How can this happen?

Reply:
Goody catch - this is a leftover comment from previous changes where I was modifying tree update operation.